dev/tasks/docker-tests/azure.linux.yml

```diff
@@ -45,6 +45,15 @@ jobs:
       git -C arrow submodule update --init --recursive
     displayName: Clone arrow
 
+  # only needed for r-lto job to use until https://github.com/wch/r-debug/pull/22 is merged
+  - script: |
+      git clone https://github.com/nealrichardson/r-debug.git
+      git checkout lto
+      cd r-debug
+      ./buildall.sh
```

Comment:
A possibly faster way to do this is to add a new job like for the r-sanitizer, and add dockerfile like https://github.com/nealrichardson/r-debug/blob/lto/r-debug-5/Dockerfile:

```
FROM wch1/r-debug

# RDlto: R built with --enable-lto
RUN wget https://raw.githubusercontent.com/nealrichardson/r-debug/lto/r-devel/buildR.sh
RUN buildR.sh lto
RUN RDlto -q -e 'install.packages("pak", repos = "https://r-lib.github.io/p/pak/dev")'
RUN RDlto -q -e 'pak::pkg_install(c("devtools", "Rcpp", "roxygen2", "testthat", "memoise", "rmarkdown"))'
```

Reply:
Actually, turns out R-hub has a docker image already: https://github.com/r-hub/rhub-linux-builders/blob/master/debian-gcc-devel-lto/Dockerfile